browser/brave_tab_helpers.cc

```diff
@@ -131,19 +131,19 @@ void AttachTabHelpers(content::WebContents* web_contents) {
   brave_rewards::RewardsTabHelper::CreateForWebContents(web_contents);
 
 #if BUILDFLAG(ENABLE_AI_CHAT)
-  if (ai_chat::features::IsAIChatEnabled()) {
     content::BrowserContext* context = web_contents->GetBrowserContext();
-    auto skus_service_getter = base::BindRepeating(
-        [](content::BrowserContext* context) {
-          return skus::SkusServiceFactory::GetForContext(context);
-        },
-        context);
-    ai_chat::AIChatTabHelper::CreateForWebContents(
-        web_contents,
-        g_brave_browser_process->process_misc_metrics()->ai_chat_metrics(),
-        skus_service_getter, g_browser_process->local_state(),
-        std::string(version_info::GetChannelString(chrome::GetChannel())));
-  }
+    if (ai_chat::IsAIChatEnabled(context)) {
```

Comment:
this exact change means that the feature state is not global anymore, but browser-state dependent. Dependencies have changed, so that means you also need to change the code layout.

Reply:
I will leave out the `IsAIChatEnabled(content::BrowserContext* context)`, since that function is only introduced for accommodating one line of `IsTor()` check. Context checking should be done in different PR.

Reply:
And it will be fixed in https://github.com/brave/brave-browser/issues/35835